src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationOptions.cs

```diff
@@ -18,10 +20,11 @@ namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
     /// Options used to configure the behavior of an Azure App Configuration provider.         
     /// If neither <see cref="Select"/> nor <see cref="SelectSnapshot"/> is ever called, all key-values with no label are included in the configuration provider.
     /// </summary>
-    public class AzureAppConfigurationOptions
+    public class AzureAppConfigurationOptions : IDisposable
```

Comment:
When is dispose called on this in a customer app?

Reply:
Missed this, looks like we could call `_options.Dispose` in `AzureAppConfigurationProvider.Dispose`?

Reply:
Whoever creates a disposable object should be the one to dispose it. In our case `AzureAppConfigurationOptions` is created internally and doesn't make sense to be disposed where it's created.

So I'm not sure we should be doing it this way at all. I'm leaning towards us doing this via a policy instead of passing an HttpClient.

Reply:
I've looked into it, and it seems like the [`RetryOptions.NetworkTimeout`](https://learn.microsoft.com/en-us/dotnet/api/azure.core.retryoptions.networktimeout?view=azure-dotnet) property also throws a [`TaskCanceledException`](https://stackoverflow.com/questions/78649680/how-to-change-networktimeout-for-defaultazurecredential#:~:text=To%20resolve%20the%20Network%20Timeout%20error%2C%20use%20Retry,and%20used%20below%20code%20to%20generate%20access%20token.) on timeout. It's difficult to find actual documentation on this property, but it appears to function the same.

Reply:
Did you check the exception composition? Because right now a check is being added for TaskCanceled with Timeout as an inner exception.

Reply:
Tested it locally, looks like it's not a `TimeoutException` as the inner exception, just another `TaskCanceledException`. I'll update this

Reply:
Is there a test that can be added to validate?

Reply:
I've added a test to check that the provider fails over and checks the exception type if there's no replica for failover.